test/unit/stats.js

```diff
@@ -0,0 +1,793 @@
+        // assert.equal(err.propertyValue, req.params.student_id);
+        assert.equal(err.status, 400);
+        done();
+      });
+    });
+
+    // test for student_id not in database: what should it do?
+    xit('should give an error if the student_id is not in the database',
+    function(done) {
+      var req = {
+        params: {
+          student_id: 48394234
+        }
+      };
+
+      var promise = stats.getStats(req);
+      promise.catch(function(err) {
+        assert.equal(err.message,
+        'Given student_id is not found in the database.');
+        assert.equal(err.status, 400);
+        done();
+      });
+    });
+  });
+
+  describe('getStat(req)', function() {
+    xit('should get a specific stat', function(done) {
+      var req = {
+        params: {
+          // The student_id is contained in the request
+          stat_id: 4
+        }
+      };
+
+      var promise = stats.getStat(req);
+
+      promise.then(function(data) {
+        assert.deepEqual(data, [fakeStat4]);
+        assert.length(data, 1);
+        done();
+      });
+    });
+
+    xit('should give an error if the stat_id is negative',
+    function(done) {
+      var req = {
+        params: {
+          stat_id: -5
+        }
+      };
+
+      var promise = stats.getStat(req);
+      promise.catch(function(err) {
+        assert.equal(err.message,
+        'Given stat_id is of invalid format (e.g. not an integer or' +
+        ' negative)');
+
+        // assert.equal(err.name, 'InvalidArgumentError');
+        // assert.equal(err.propertyName, 'stat_id');
+        // assert.equal(err.propertyValue, req.params.stat_id);
+        assert.equal(err.status, 400);
+        done();
+      });
+    });
+
+    xit('should give an error if the stat_id is not an integer',
+    function(done) {
+      var req = {
+        params: {
+          stat_id: 'that stat'
+        }
+      };
+
+      var promise = stats.getStat(req);
+      promise.catch(function(err) {
+        assert.equal(err.message,
+        'Given stat_id is of invalid format (e.g. not an integer or' +
+        ' negative)');
+
+        // assert.equal(err.name, 'InvalidArgumentError');
+        // assert.equal(err.propertyName, 'stat_id');
+        // assert.equal(err.propertyValue, req.params.stat_id);
+        assert.equal(err.status, 400);
+        done();
+      });
+    });
+
+    xit('should give an error if the stat_id is not in the database',
+    function(done) {
+      var req = {
+        params: {
+          stat_id: 9999999
+        }
+      };
+
+      var promise = stats.getStat(req);
+      promise.catch(function(err) {
+        assert.equal(err.message,
+        'Given stat_id is not found in the database');
+        assert.equal(err.status, 400);
+        done();
+      });
+    });
+  });
+
+  describe('createStat(req)', function() {
+    // Post a new row of stats into stats database
+    xit('should add new stats to the database', function(done) {
+      var req = {
```

Comment:
Note that the request for create stat should look like:
req = { 
params: {
event_id: #,
student_id: #
},
body {
height: #,
weight: ###,
pacer: ###
}
}